tools/font-loader-lib/index.js

```diff
@@ -0,0 +1,34 @@
+const fetchFonts = (window, document) => {
+    const head = document.querySelector('head');
+    const useFont = font => {
+        if (font.css) {
+            const style = document.createElement('style');
+            style.innerHTML = font.css;
+            head.appendChild(style);
+        }
+    };
+
+    const loadFonts = () => {
+        const iframe = document.createElement('iframe');
+        iframe.src = 'https://theguardian.com/font-loader';
+        iframe.classList = 'guardianFontLoader';
+        // add iframe and wait for message
+        iframe.style.display = 'none';
+        window.addEventListener('message', e => {
+            if (
+                e &&
+                e.data &&
+                e.data.name &&
+                e.data.name === 'guardianFonts' &&
+                e.data.fonts &&
+                e.source === iframe.contentWindow
+            ) {
+                e.data.fonts.forEach(useFont);
+            }
+        });
+        document.body.appendChild(iframe);
+    };
+    loadFonts();
```

Comment:
This `loadFonts` call should only ever be executed when the DOM has been parsed. This module could feasibly be bundled in a script that's executed before this has happened, really we should wrap `loadFonts()` in this:

```JS
if (document.readyState === "loading") {  
  // Loading hasn't finished yet
  document.addEventListener("DOMContentLoaded",  loadFonts);
} else {  
  // DOMContentLoaded has already fired
  loadFonts();
}
```